runtime-api/src/main/scala/io/coral/actors/CoralActor.scala

```diff
@@ -177,35 +177,40 @@ trait CoralActor extends Actor with ActorLogging {
 	}
 
 	def jsonData: Receive = {
-		case json: JObject =>
-			val stage = trigger(json)
-			val r = stage.run
-
-			r.onSuccess {
-				case Some(_) => transmit(emit(json))
-				case None => log.warning("some variables are not available")
-			}
-
-			r.onFailure {
-				case _ => //log.warning("oh no, timeout or other serious exceptions!")
-			}
-		case Request(json) =>
-			val s = sender
-			val stage = trigger(json)
-			val r = stage.run
-
-			r.onSuccess {
-				case Some(_) =>
-					val result = emit(json)
-					transmit(result)
-				    if (result != JNothing) s ! result
-
-				case None => log.warning("some variables are not available")
-			}
-
-			r.onFailure {
-				case e => println(e) //log.warning("oh no, timeout or other serious exceptions!")
-			}
+        case json: JObject =>
+            val stage = trigger(json)
+            val r = stage.run
+
+            r.onSuccess {
+                case Some(_) => transmit(emit(json))
+                case None => log.warning("some variables are not available")
+            }
+
+            r.onFailure {
+                case _ =>
+            }
+        case Shunt(json) =>
+            val s = sender
+            val stage = trigger(json)
+            val r = stage.run
+
+            r.onSuccess {
+                case Some(_) =>
+                    val result = emit(json)
+                    transmit(result)
+                    if (result != JNothing) s ! result
+
+                case None => log.warning("some variables are not available")
+            }
+
+            r.onFailure {
+                case e => println(e)
+            }
+        case Trigger(json) =>
```

Comment:
We could add more detailed exception handling but for now, this was also not present in the original trigger call, only if the asynchronous call itself has succeeded or not.